src/Language/PureScript/Sugar/TypeClasses/Deriving.hs

```diff
@@ -404,6 +421,47 @@ deriveOrd mn ds tyConNm = do
       $ decomposeRec rec
     toOrdering l r _ = ordCompare l r
 
+deriveNewtype
+  :: forall m
+   . (MonadError MultipleErrors m, MonadSupply m)
+  => ModuleName
+  -> [Declaration]
+  -> ProperName 'TypeName
+  -> Type
+  -> m ([Declaration], Type)
+deriveNewtype mn ds tyConNm unwrappedTy = do
+  checkIsWildcard unwrappedTy
+  go =<< findTypeDecl tyConNm ds
+  where
+
+  go :: Declaration -> m ([Declaration], Type)
+  go (DataDeclaration Data name _ _) =
+    throwError . errorMessage $ CannotDeriveNewtypeForData name
+  go (DataDeclaration Newtype name _ dctors) = do
+    checkNewtype name dctors
+    let (ctorName, [ty]) = head dctors
```

Comment:
Could do, but if you have an invalid `newtype` you'll get an error about an un-derivable class before you get the error that the `newtype` is invalid (since the check is made during type checking), which seems... not very useful.